### PR TITLE
Move BufferedInput and InputStreams to dwio::common

### PR DESCRIPTION
**Pull request by yingsu00**
BufferedInput, InputStream, SeekableInputStream and their child classes
will be used by both ORC/DWRF and Parquet readers. This commit moves
BufferedInput, SeekableInputStream and their child classes from
dwio::dwrf to dwio::common::io, and InputStream and its child classes
from dwio::common to dwio::common::io. Note that PagedInputStream was
not moved for now, because it is tightly coupled with ORC/DWRF format
and will only be used by the ORC/DWRF reader.

This PR depends on https://github.com/facebookincubator/velox/pull/1754

### CODE REVIEW

File: velox/connectors/hive/FileHandle.h [hunk 1]
```diff
@@ -29,11 +29,11 @@
 #include <memory>
 #include <string>
 
+#include "dwio/common/InputStream.h"
```
Comment (zzhao0) [inline velox/connectors/hive/FileHandle.h:32]: Let's keep the full path here. Otherwise it breaks the build system on Meta side

**Review by zzhao0 — COMMENTED** [review]
LGTM. I will import once the base commit is merged and will have to make some fixes

File: velox/connectors/hive/FileHandle.h (hunk 1 above)
Comment (yingsu00) [reply velox/connectors/hive/FileHandle.h]: @zzhao0 Thanks for reviewing the PR. I actually had this fixed before but the change was lost. Now the new commit contains the fix.

File: velox/connectors/hive/HiveConnector.cpp [hunk 2]
```diff
@@ -17,9 +17,9 @@
 
 #include <memory>
 
-#include "velox/dwio/common/InputStream.h"
+#include "dwio/common/CachedBufferedInput.h"
```
Comment (zzhao0) [inline velox/connectors/hive/HiveConnector.cpp:20]: let's use velox/dwio/common/...

File: velox/dwio/common/CacheInputStream.cpp [hunk 3]
```diff
@@ -14,23 +14,23 @@
  * limitations under the License.
  */
 
-#include "velox/dwio/dwrf/common/CacheInputStream.h"
+#include "CacheInputStream.h"
```
Comment (zzhao0) [inline velox/dwio/common/CacheInputStream.cpp:17]: same, full path like before

File: velox/dwio/common/CacheInputStream.h [hunk 4]
```diff
@@ -16,23 +16,24 @@
 
 #pragma once
 
+#include "InputStream.h"
```
Comment (zzhao0) [inline velox/dwio/common/CacheInputStream.h:19]: full path again

File: velox/dwio/common/InputStream.cpp [hunk 5]
```diff
@@ -320,7 +311,4 @@ VELOX_REGISTER_INPUT_STREAM_METHOD_DEFINITION(
     FileInputStream,
     fileInputStreamFactory)
 
-} // namespace common
-} // namespace dwio
-} // namespace velox
-} // namespace facebook
+} // namespace facebook::velox::dwio::common
```
Comment (zzhao0) [inline velox/dwio/common/InputStream.cpp:314]: add new line

File: velox/dwio/common/SeekableInputStream.h [hunk 6]
```diff
@@ -16,14 +16,15 @@
...
-#include "velox/dwio/common/InputStream.h"
-#include "velox/dwio/dwrf/common/wrap/dwrf-proto-wrapper.h"
 #include "velox/dwio/dwrf/common/wrap/zero-copy-stream-wrapper.h"
 
 #include <vector>
 
-namespace facebook::velox::dwrf {
+namespace facebook::velox::dwio::common {
+
+using namespace dwio::common;
```
Comment (zzhao0) [inline velox/dwio/common/SeekableInputStream.h:27]: let's avoid this in .h file

File: velox/dwio/common/SeekableInputStream.h [hunk 7]
```diff
@@ -16,14 +16,15 @@
 
 #pragma once
 
+#include "InputStream.h"
 #include "velox/dwio/common/DataBuffer.h"
-#include "velox/dwio/common/InputStream.h"
-#include "velox/dwio/dwrf/common/wrap/dwrf-proto-wrapper.h"
 #include "velox/dwio/dwrf/common/wrap/zero-copy-stream-wrapper.h"
```
[the remark this answers is not in the document]
Comment (yingsu00) [reply velox/dwio/common/SeekableInputStream.h]: @zzhao0 I saw this header contains some additional conditions, so I moved it to velox/dwio/common/wrap/zero-copy-stream-wrapper.h. Let me know if you want the direct google include instead.

File: velox/dwio/dwrf/test/E2EFilterTest.cpp [hunk 8]
```diff
@@ -22,7 +22,6 @@ using namespace facebook::velox::dwrf;
 using namespace facebook::velox;
 using namespace facebook::velox::common;
 
-using dwio::common::MemoryInputStream;
```
Comment (yingsu00) [inline velox/dwio/dwrf/test/E2EFilterTest.cpp:25]: FYI MemoryInputStream was actually not used. Just cleaning up.

**Review by yingsu00 — COMMENTED** [review]
@zzhao0 Thank you very much for your careful review. I have addressed all comments, and would appreciate your review again.

**Pull request closed without merging**